### ISSUE TITLE
Output differ between keras prediction and DL4j(model from keras trained) prediction

### ISSUE THREAD

**Issue by Arkariian** [deeplearning4j/deeplearning4j#6108]
hi!
I have this net trained with keras with 99% accuracy. When i predict in keras the result is one and DL4j is other.
In keras: [[1.8116718e-05 9.9998188e-01]]
IN DL4J: [[9.9999987e-01 0.0000278e-12]]

`    
    
    #model.add(Lambda())  #resize images to allow for easy computation
    #model.add(Conv2D(16,(5,5),padding = 'same',activation='relu'))
    #model.add(MaxPooling2D(pool_size=(3,3)))
    #model.add(Dropout(0.25))
    model = Sequential()
    inputShape = (height, width, depth)
    model.add(ZeroPadding2D((1, 1), input_shape=inputShape))
    model.add(Conv2D(32,(5,5),padding='same',activation='relu'))
    model.add(MaxPooling2D(pool_size=(2,2)))
    #model.add(Dropout(0.25))
   
    model.add(Conv2D(64,(3,3),padding='same',activation='relu'))
    model.add(MaxPooling2D(pool_size=(2,2)))
    #model.add(Dropout(0.25))
    
    model.add(Conv2D(128,(3,3),padding='same',activation='relu'))
    model.add(MaxPooling2D(pool_size=(2,2)))
    #model.add(Dropout(0.25))
    
    
    model.add(Flatten())
    model.add(Dropout(0.5))
              
    model.add(Dense(activation='relu',units=512))
    model.add(BatchNormalization())
    model.add(Dropout(0.5))
    model.add(Dense(256, name='dense2')) 
    model.add(Activation('relu'))
    model.add(Dropout(0.5))
    model.add(Dense(activation='softmax',units = classes))
    
    
    model.compile(loss=keras.losses.categorical_crossentropy,
              optimizer=Adam(),
              metrics=['accuracy'])`

my prediction method in python is:
`test_image = image.load_img(file_test_path, target_size=(64, 85), grayscale=True)
test_image = image.img_to_array(test_image)
test_image = np.expand_dims(test_image, axis=0)
result = classifier.predict(test_image)
print(result)`

and in DL4j is:
`NativeImageLoader loader = new NativeImageLoader(64, 85, 1);
        INDArray imageMatrix = loader.asMatrix(image);
            Float validPercentaje = model.output(imageMatrix).getFloat(1) * 100;`

i don't know what is wrong or if is a bug. my snapshot version is 1.0.0-beta

Thank you!

**Comment by maxpumperla** [comment]
@Arkariian I need your model to reproduce this, but: before I do anything please test both models with artificial data that don't involve your manual preprocessing steps (just to make sure that's not the issue)

**Comment by Arkariian** [comment]
@maxpumperla yes i try both models with random data(but the same on each model), this is the code en DL4j i can't attach the model because is a big file (55mb). how can i do it?

`public class Model {


	private static Logger logger = Logger.getLogger(Model.class);

    private final static String SIGNATURE_MODEL_PATH = "signature.model.path";
    private final static String SIGNATURE_MODEL_IMAGE_WIDTH = "signature.model.image.width";
    private final static String SIGNATURE_MODEL_IMAGE_HEIGHT = "signature.model.image.height";
    private final static String SIGNATURE_MODEL_IMAGE_DEPTH = "signature.model.image.depth";
    private final static String SIGNATURE_MODEL_SIGNATURE_PERCENTAGE_POSITION = "signature.model.signature.percentage.position";
	
    
    private static String modelPath;
	private static int imageWidth;
	private static int imageHeight;
	private static int imageDepth;
	private static int percentagePosition;

	private static Model model_instance = null;
	private static MultiLayerNetwork model;
//	private static INDArray imageMatrix = null;
//	private static INDArray result = null;
	private static NativeImageLoader loader = null;
//	private static Float validPercentaje = null;

	/**
	 * Constructor de la clase
	 * @throws Exception 
	 */
	private Model() throws Exception{
		try {
			
			System.out.println("Model ctor Inicio.");
			logger.debug("Model ctor Inicio.");
			Properties properties = new Properties();
			properties.load(BesignRestService.class.getResourceAsStream("/besign.properties"));
			modelPath = (properties.getProperty(SIGNATURE_MODEL_PATH) != null) && (!properties.getProperty(SIGNATURE_MODEL_PATH).isEmpty()) ? properties.getProperty(SIGNATURE_MODEL_PATH) : null;
			imageWidth = (properties.getProperty(SIGNATURE_MODEL_IMAGE_WIDTH) != null) && (!properties.getProperty(SIGNATURE_MODEL_IMAGE_WIDTH).isEmpty()) ? Integer.valueOf(properties.getProperty(SIGNATURE_MODEL_IMAGE_WIDTH)) : null;
			imageHeight = (properties.getProperty(SIGNATURE_MODEL_IMAGE_HEIGHT) != null) && (!properties.getProperty(SIGNATURE_MODEL_IMAGE_HEIGHT).isEmpty()) ? Integer.valueOf(properties.getProperty(SIGNATURE_MODEL_IMAGE_HEIGHT)) : null;
			imageDepth = (properties.getProperty(SIGNATURE_MODEL_IMAGE_DEPTH) != null) && (!properties.getProperty(SIGNATURE_MODEL_IMAGE_DEPTH).isEmpty()) ? Integer.valueOf(properties.getProperty(SIGNATURE_MODEL_IMAGE_DEPTH)) : null;
			percentagePosition = (properties.getProperty(SIGNATURE_MODEL_SIGNATURE_PERCENTAGE_POSITION) != null) && (!properties.getProperty(SIGNATURE_MODEL_SIGNATURE_PERCENTAGE_POSITION).isEmpty()) ? Integer.valueOf(properties.getProperty(SIGNATURE_MODEL_SIGNATURE_PERCENTAGE_POSITION)) : null;
			model = KerasModelImport.importKerasSequentialModelAndWeights(modelPath);
			loader = new NativeImageLoader(imageHeight, imageWidth, imageDepth);
			logger.debug("Model ctor Fin.");
			System.out.println("Model ctor Fin.");
		} catch (Exception e) {
			System.out.println("Model ctor No se ha podido cargar el modelo: " + e.toString());
			logger.error("Model ctor No se ha podido cargar el modelo: " + e.toString());
			throw e;
		}
	}
	
	/**
	 * Método que devuelve la instancia del singleton
	 * @return Objeto de la clase Model
	 * @throws Exception 
	 */
    public static Model getInstance() throws Exception
    {
    	System.out.println("Model.getInstance Inicio.");
    	logger.debug("Model.getInstance Inicio.");
		if (model_instance == null) {
        	System.out.println("Model.getInstance Crea modelo.");
        	logger.info("Model.getInstance Crea modelo.");
        	model_instance = new Model();
        }
        System.out.println("Model.getInstance Fin.");
    	logger.debug("Model.getInstance Fin.");
        return model_instance;
    }

    /**
     * Metodo que realiza la predicción sobre el modelo con la imagen recibida para
     * determinar si esta contiene o no una firma válida.
     * @param image imagen que contiene la firma a determinar como válida o no
     * @return porcentaje de acierto de firma valida.
     * @throws Exception
     */
    public Float predictionPercentaje(InputStream image) throws Exception {
    	logger.debug("Model.predictionPercentaje Inicio.");
    	Float validPercentaje=-1f;
    	if (image != null) {
    		INDArray imageMatrix = loader.asMatrix(image);
    		if (imageMatrix != null) {
		    	logger.debug("Model.predictionPercentaje Cargada la imagen como matriz.");
		        //Float validPercentaje = model.output(imageMatrix).getFloat(1) * 100;//posición de firma
		    	
		    	INDArray result = model.output(imageMatrix);
		    	logger.debug("Model.predictionPercentaje Creada la predicción.");
		    	validPercentaje = result.getFloat(percentagePosition) * 100;//posición de firma
		    	result.cleanup();
		    	result = null;
		    	imageMatrix.cleanup();
		    	imageMatrix = null;
    		} else {
        		logger.error("Model.predictionPercentaje Error la matriz de la imagen es null.");
    		}
    	} else {
    		logger.error("Model.predictionPercentaje Error la imagen es null.");
    	}
    	logger.warn("Model.predictionPercentaje Resultado: " + validPercentaje);
    	logger.debug("Model.predictionPercentaje Fin.");
        return validPercentaje;
    }

}
`

**Comment by maxpumperla** [comment]
@Arkariian can you put it on gdrive or so? otherwise hit me at 

max at skymind io

**Comment by Arkariian** [comment]
@maxpumperla 
this is the gdrive link [https://drive.google.com/file/d/1HDnie7KmO3ozJDJWMSGw28mt5rUuNQWH/view?usp=sharing](url)

**Comment by maxpumperla** [comment]
cool, thanks. what's the input shape for this model?

**Comment by Arkariian** [comment]
Input shape is 64x85 with depth 1. 

 I dont know if the problem is numpy array to INDarray or de model

**Comment by Arkariian** [comment]
@maxpumperla i check that the input is exactly the same in both cases, so i don't know why is different on the output, if is only a keras import, i 'll add a file with an input array.
[2_to_array.txt](https://github.com/deeplearning4j/deeplearning4j/files/2274083/2_to_array.txt)

**Comment by Arkariian** [comment]
@maxpumperla any news?

**Comment by maxpumperla** [comment]
@Arkariian I'll let you know. We don't give time-lines in open source.
